src/main/java/uk/gov/pay/adminusers/service/NewUserExistingServiceInviteCompleter.java

```diff
@@ -0,0 +1,77 @@
+
+import javax.ws.rs.WebApplicationException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static uk.gov.pay.adminusers.service.AdminUsersExceptions.conflictingEmail;
+import static uk.gov.pay.adminusers.service.AdminUsersExceptions.internalServerError;
+import static uk.gov.pay.adminusers.service.AdminUsersExceptions.inviteLockedException;
+import static uk.gov.pay.adminusers.service.AdminUsersExceptions.notFoundInviteException;
+
+public class NewUserExistingServiceInviteCompleter extends InviteCompleter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NewUserExistingServiceInviteCompleter.class);
+    
+    private final InviteDao inviteDao;
+    private final UserDao userDao;
+    private final LinksBuilder linksBuilder;
+
+    @Inject
+    public NewUserExistingServiceInviteCompleter(InviteDao inviteDao, UserDao userDao, LinksBuilder linksBuilder) {
+        super();
+        this.inviteDao = inviteDao;
+        this.userDao = userDao;
+        this.linksBuilder = linksBuilder;
+    }
+
+    @Override
+    @Transactional
+    public Optional<InviteCompleteResponse> complete(String inviteCode) {
+        return inviteDao.findByCode(inviteCode)
+                .map(inviteEntity -> {
+                    if (inviteEntity.isExpired() || inviteEntity.isDisabled()) {
+                        throw inviteLockedException(inviteEntity.getCode());
+                    }
+                    if (userDao.findByEmail(inviteEntity.getEmail()).isPresent()) {
+                        throw conflictingEmail(inviteEntity.getEmail());
+                    }
+                    if (inviteEntity.getType() != InviteType.NEW_USER_INVITED_TO_EXISTING_SERVICE) {
+                        throw internalServerError(format("Attempting to complete a 'new user, existing service' invite for an invite of type '%s'", inviteEntity.getCode()));
+                    }
+                    
+                    UserEntity userEntity = inviteEntity.mapToUserEntity();
+                    userDao.persist(userEntity);
+                    inviteEntity.setDisabled(Boolean.TRUE);
+                    inviteDao.merge(inviteEntity);
+                    
+                    String serviceIds = userEntity.getServicesRoles().stream()
+                            .map(serviceRole -> serviceRole.getService().getExternalId())
+                            .collect(Collectors.joining(", "));
+
+                    LOGGER.info("User created successfully from invitation [{}] for services [{}]", userEntity.getExternalId(), serviceIds);
+
+                    Invite invite = linksBuilder.addUserLink(userEntity.toUser(), inviteEntity.toInvite());
+                    InviteCompleteResponse response = new InviteCompleteResponse(invite);
+                    response.setUserExternalId(userEntity.getExternalId());
+
+                    return Optional.of(response);
```

Comment:
Looks like we don't need to return an Optional, as it will always have a value. If the other completers don't need to return an Optional, we could refactor this in a following PR?

Reply:
One of the other completers does return an Optional, but I think it makes more sense for it to work that same way that my new completer does.  I'll refactor it as part of this PR, so that we don't have inconsistent code in the interim.

Reply:
On second thoughts, it means changing the interface...I'll go along with your suggestion of using a separate PR. :)